r/src/bpcells-cpp/matrixIterators/H5DenseMatrixWriter.h

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 BPCells contributors
+// 
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#pragma once
+
+#include "../arrayIO/array_interfaces.h"
+#include "../arrayIO/bp128.h"
+#include "../arrayIO/hdf5.h"
+#include "../utils/filesystem_compat.h"
+#include "MatrixIterator.h"
+#include "OrderRows.h"
+
+namespace BPCells {
+
+template <typename T> class H5DenseMatrixWriter : public MatrixWriter<T> {
+  private:
+    HighFive::DataSet dataset;
+    HighFive::DataType datatype = HighFive::create_datatype<T>();
+    uint64_t max_capacity;
+    bool row_major;
+
+    static HighFive::DataSet createH5Matrix(
+        HighFive::File file,
+        std::string dataset_path,
+        uint64_t nrow,  // nrow of input matrix
+        uint64_t ncol,  // ncol of input matrix
+        bool row_major,
+        uint64_t chunk_size,
+        uint32_t gzip_level
+    ) {
+      HighFive::SilenceHDF5 s;
+      // Create a dataspace with initial shape and max shape
+      uint64_t nrow_h5 = nrow;
+      uint64_t ncol_h5 = ncol;
+      if (row_major) {
+          nrow_h5 = ncol;
+          ncol_h5 = nrow;
+      }
+      HighFive::DataSpace dataspace({nrow_h5, ncol_h5}, {HighFive::DataSpace::UNLIMITED, HighFive::DataSpace::UNLIMITED});
```

Comment:
I don't think we need to make these datasets resizable, so I'd suggest this:
```suggestion
      HighFive::DataSpace dataspace({nrow_h5, ncol_h5});
```